libs/ledger-live-common/src/families/internet_computer/bridge/account.ts

```diff
@@ -0,0 +1,262 @@
+      (await validateAddress(address)).isValid
+    ) {
+      if (t.useAllAmount) {
+        t.amount = a.spendableBalance.minus(t.fees);
+      }
+    }
+  }
+
+  // log("debug", "[prepareTransaction] finish fn");
+  return t;
+};
+
+const getTransactionStatus = async (
+  a: Account,
+  t: Transaction
+): Promise<TransactionStatus> => {
+  const errors: TransactionStatus["errors"] = {};
+  const warnings: TransactionStatus["warnings"] = {};
+
+  const { balance } = a;
+  const { address } = getAddress(a);
+  const { recipient, useAllAmount } = t;
+  let { amount } = t;
+
+  if (!recipient) errors.recipient = new RecipientRequired();
+  else if (!(await validateAddress(recipient)).isValid)
+    errors.recipient = new InvalidAddress();
+  else if (recipient.toLowerCase() === address.toLowerCase())
+    errors.recipient = new InvalidAddressBecauseDestinationIsAlsoSource();
+
+  if (!(await validateAddress(address)).isValid)
+    errors.sender = new InvalidAddress();
+
+  if (!validateMemo(t.memo).isValid) errors.transaction = new InvalidMemoICP();
+
+  // This is the worst case scenario (the tx won't cost more than this value)
+  const estimatedFees = t.fees;
+
+  let totalSpent: BigNumber;
+
+  if (useAllAmount) {
+    totalSpent = a.spendableBalance;
+    amount = totalSpent.minus(estimatedFees);
+    if (amount.lte(0) || totalSpent.gt(balance)) {
+      errors.amount = new NotEnoughBalance();
+    }
+  } else {
+    totalSpent = amount.plus(estimatedFees);
+    if (amount.eq(0)) {
+      errors.amount = new AmountRequired();
+    } else if (totalSpent.gt(a.spendableBalance)) {
+      errors.amount = new NotEnoughBalance();
+    }
+  }
+
+  // log("debug", "[getTransactionStatus] finish fn");
+
+  return {
+    errors,
+    warnings,
+    estimatedFees,
+    amount,
+    totalSpent,
+  };
+};
+
+const estimateMaxSpendable = async ({
+  account,
+  transaction,
+}: {
+  account: AccountLike;
+  parentAccount?: Account | null | undefined;
+  transaction?: Transaction | null | undefined;
+}): Promise<BigNumber> => {
+  const balance = account.balance;
+
+  let maxSpendable = balance.minus(transaction?.fees ?? getEstimatedFees());
+
+  if (maxSpendable.isLessThan(0)) {
+    maxSpendable = new BigNumber(0);
+  }
+
+  return maxSpendable;
+};
+
+const signOperation: SignOperationFnSignature<Transaction> = ({
+  account,
+  deviceId,
+  transaction,
+}): Observable<SignOperationEvent> =>
+  withDevice(deviceId)(
+    (transport) =>
+      new Observable((o) => {
+        async function main() {
+          // log("debug", "[signOperation] start fn");
+
+          const { recipient, useAllAmount } = transaction;
+          let { amount } = transaction;
+          const { id: accountId, balance, xpub } = account;
+          const { address, derivationPath } = getAddress(account);
+          const fee = transaction.fees;
+          if (useAllAmount) amount = balance.minus(fee);
+
+          transaction = { ...transaction, amount };
```

Comment:
This belongs in `prepareTransaction` (I think you already do it there).
The role of `prepareTransaction` is to make any adjustment to the transaction while it's being built, then the transaction should be final during the `signOperation` step.

Reply:
Makes perfect sense, nice catch. Fixed!